### PR TITLE
Create APIReview_UAString

### PR DESCRIPTION
**Pull request by jasonstephen15**
*NOTE*: THIS BRANCH IS A DRAFT FOR REVIEW PURPOSES. PLEASE REFER TO 'api-useragent' for official api review.

### CODE REVIEW

File: specs/APIReview_UAString [hunk 1]
```diff
@@ -0,0 +1,149 @@
+The purpose of this spec is to describe a new WebView2 feature - Support for UA Strings - and its APIs.
```
Comment (david-risney) [inline specs/APIReview_UAString:1]: This document is missing the APIs and sample code.

Comment (david-risney) [inline specs/APIReview_UAString:1]: For an initial spec review you review with me and you do a PR from a branch like api-uastring-draft to api-uastring. After that you do a PR from api-uastring to master and that we'll use for review with API board and customers on github

**Review by david-risney — CHANGES_REQUESTED** [review]

File: specs/APIReview_UAString (hunk 1 above)
Comment (jasonstephen15) [reply specs/APIReview_UAString]: Ah, my bad, this was meant to be the api-uastring-draft, after we get the review, maybe I can create another branch 'api-useragentstring' as the one we can review w/ api board & customers @david-risney - would this be ok?

Comment (david-risney) [reply specs/APIReview_UAString]: The name of the branch isn't a big deal. Its more important to not go straight to master:

PR1, branch A -> branch B, we review together
PR2, branch B -> master, we review with API review board

Can you change this PR to not go to master?

File: specs/APIReview_UAString.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        .Get(),
+    &m_navigationStartingToken);
+``` 
+
+## .NET and WinRT
+
+```c #
+private void SetUserAgent(CoreWebView2 sender, CoreWebView2UserAgentArgs e) {
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.UserAgent = "Mozilla/5.0 (Linux; Android 8.0.0; SM-G960F Build/R16NW) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.84 Mobile Safari/537.36";
```
Comment (david-risney) [inline specs/APIReview_UAString.md:57]: We probably don't want to embed an actual UA here. How about we use a static constant string that we exclude the defintion of.

**Review by david-risney — CHANGES_REQUESTED** [review]

File: specs/APIReview_UAString (hunk 1 above)
Comment (jasonstephen15) [reply specs/APIReview_UAString]: Took care of it, sorry for the confusion!

File: specs/APIReview_UAString.md (hunk 2 above)
Comment (maurawinstanley) [reply specs/APIReview_UAString.md:57]: Should that be passes in as an arg or in the function body: 
static const std::string updatedUA;
settings.UserAgent = updatedUA;

File: specs/APIReview_UAString.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        })
+        .Get(),
+    &m_navigationStartingToken);
+``` 
+
+## .NET and WinRT
+
+```c #
+private void SetUserAgent(CoreWebView2 sender, CoreWebView2UserAgentArgs e) {
+    static const std::string updatedUA;
```
Comment (david-risney) [inline specs/APIReview_UAString.md:59]: This is C#. std::string is not a thing in C#. Maybe you could do a method like GetMobileUserAgent() and not defined it instead?

File: specs/APIReview_UAString.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                    "AppleWebKit/537.36 (KHTML, like Gecko) "
+                    "Chrome/62.0.3202.84 Mobile Safari/537.36";
+                CHECK_FAILURE(settings->put_UserAgent(mobile_ua)); 
+            } else {
+                // Change the user agent back to desktop
+                LPCWSTR desktop_ua =
+                    L"Mozilla/5.0 (Linux; Android 8.0.0; SM-G960F Build/R16NW) "
+                    "AppleWebKit/537.36 (KHTML, like Gecko) "
+                    "Chrome/62.0.3202.84 Safari/537.36";
+                CHECK_FAILURE(settings->put_UserAgent(desktop_ua));
```
Comment (david-risney) [inline specs/APIReview_UAString.md:47]: I was imagining similarly for this that we don't want hardcoded user agents in the sample code. These can change over time and we don't want people copy/pasting them into their code. Maybe just a method like in the other case GetMobileUserAgent() GetDesktopUserAgent() ?

**Review by david-risney — CHANGES_REQUESTED** [review]

File: specs/APIReview_UAString.md (hunk 2 above)
Comment (david-risney) [reply specs/APIReview_UAString.md]: Sorry didn't see this question. I was imagining a UA string constant defined elsewhere not shown or a method that returns it not shown, or a parameter would also work.

**Review by david-risney — APPROVED** [review]

**Pull request merged**